processing/src/main/java/io/druid/segment/IndexMaker.java

```diff
@@ -796,6 +806,48 @@ private static void makeDimColumns(
     progress.stopSection(dimSection);
   }
 
+  private static class NullsAtZeroConvertingIntList extends AbstractList<Integer>
+  {
+    private final List<Integer> vals;
+
+    NullsAtZeroConvertingIntList(List<Integer> vals)
+    {
+      this.vals = vals;
+    }
+
+    @Override
+    public Integer get(int index)
+    {
+      Integer val = vals.get(index);
+      if (val == null) {
+        return 0;
+      }
+      return val + 1;
+    }
+
+    @Override
+    public int size()
+    {
+      return vals.size();
+    }
+  }
+
+  private static class AlwaysZeroConvertingIntList extends AbstractList<Integer>
```

Comment:
That's a good question. These two abstract lists are actually a bit misnamed because they are more like a Map or Function. The difference in behavior is that the current (aka prior) implementation returns 0 even for negative values